packages/framework-provider-aws/src/library/events-adapter.ts

```diff
@@ -75,7 +82,9 @@ export async function readEntityLatestSnapshot(
       `[EventsAdapter#readEntityLatestSnapshot] Snapshot found for entity ${entityTypeName} with ID ${entityID}:`,
       snapshot
     )
-    return snapshot as EventEnvelope
+    const foundSnapshot = snapshot as EventEnvelope
+    modifyEventsDecodingSortingKeys(foundSnapshot)
```

Comment:
When we read events (or snapshots), we need to "decode" de sortingKey (createdAt) to remove the suffix we added when storing it